src/Bicep.Core/Semantics/Metadata/ResourceMetadataCache.cs

```diff
@@ -87,6 +87,38 @@ public ResourceMetadataCache(SemanticModel semanticModel)
                     // Access to a module array output might be a resource metadata.
                     return this.TryAdd(module, propertyAccessSyntax.PropertyName.IdentifierName);
                 }
+                case TernaryOperationSyntax ternaryOperationSyntax:
+                {
+                        if (ternaryOperationSyntax.ConditionExpression is VariableAccessSyntax variableAccessSyntax)
+                        {
+                            var symbol = semanticModel.GetSymbolInfo(variableAccessSyntax);
+                            if (symbol is DeclaredSymbol declaredSymbol && semanticModel.Binder.TryGetCycle(declaredSymbol) is null)
+                            {
+                                if (declaredSymbol.DeclaringSyntax is VariableDeclarationSyntax variableDeclarationSyntax)
+                                {
+                                    if (variableDeclarationSyntax.Value is BooleanLiteralSyntax booleanLiteralSyntax && booleanLiteralSyntax.Value)
+                                    {
+                                        return this.TryLookup(ternaryOperationSyntax.TrueExpression);
+                                    }
+                                    else
+                                    {
+                                        return this.TryLookup(ternaryOperationSyntax.TrueExpression);
+                                    }
+                                }
+                            }
+                        }
+                        return null;
+
+                        /*//commented code below is needed for the BooleanLiteralSyntax switch statement
```

Comment:
Left this comment block on purpose because it turns out if the switch statement for BooleanLiteralSyntax simply returns null, the Parent property of the returned DeclaredResourceMetadata will always be set to the FalseExpression, even if the ConditionExpression is evaluated to be true. 

I bring this up because the Parent property doesn't seem to be used anywhere of importance, so would it be okay to do this?